tests/CSSList/DocumentTest.php

```diff
@@ -85,4 +85,66 @@ public function setContentsReplacesContentsSetInPreviousCall()
 
         self::assertSame($contents2, $this->subject->getContents());
     }
+
+    /**
+     * @return \Generator<array<array<DeclarationBlock>, DeclarationBlock, DeclarationBlock, array<DeclarationBlock>>>
+     */
+    public static function insertBeforeDataProvider(): \Generator
+    {
+
+        $bogusOne = new DeclarationBlock();
+        $bogusOne->setSelectors('.bogus-one');
+        $bogusTwo = new DeclarationBlock();
+        $bogusTwo->setSelectors('.bogus-two');
+
+        $oItem = new DeclarationBlock();
+        $oItem->setSelectors('.item');
+
+        $oSibling = new DeclarationBlock();
+        $oSibling->setSelectors('.sibling');
+
+        $oOrphan = new DeclarationBlock();
+        $oOrphan->setSelectors('.forever-alone');
+
+        yield 'insert before' => [
+            'initialContent' => [$bogusOne, $oSibling, $bogusTwo],
+            'oItem' => $oItem,
+            'oSibling' => $oSibling,
+            'expectedContent' => [$bogusOne, $oItem, $oSibling, $bogusTwo],
+            ];
+        yield 'append if not found' => [
+            'initialContent' => [$bogusOne, $oSibling, $bogusTwo],
+            'oItem' => $oItem,
+            'oSibling' => $oOrphan,
+            'expectedContent' => [$bogusOne, $oSibling, $bogusTwo, $oItem],
+            ];
+    }
+
+    /**
+     * @test
+     *
+     * @param $initialContent
+     * @param $oItem
+     * @param $oSibling
+     * @param $sPosition
+     * @param $expectedContent
```

Comment:
We can drop all those PHPDoc annotations as we already have the native type declarations.

Reply:
Got it.

Is there a possibility to automate this check?

Reply:
I remember that there was a Rector rule for this, but I don't know which one off the top of my head.